CMSSW 92X - Adding the possibility to select the PVs before the BeamSpot fit

francescobrivio opened the pull request:
This is a "reprise" of the PR #18201 (now reverted and closed).
Basically we add two parameters in PVfitter in order to be able to apply further selection on PrimaryVertices. All python cfg files have been modified to account for these new parameters (also in DQM/BeamMonitor and in Calibration/TkAlCaRecoProducers).
The default values of the parameters are set so that nothing changes with respect to the previous situation.
We added a further switch to prevent the program from crashing when the needed track collection is not found: in that case the PVselection is not applied.
I also did some "cleaning" (removed "tabs" and useless commented lines) in order to have more or less the same indendation in all the files.

sarafiorendi committed:
add possibility to cuts on # tracks, sumPt and first Vtx

francescobrivio committed:
PV selections and try/catch for cut on sumPt

francescobrivio committed:
fix PVfitter in DQM and Calibration, fixing indentations

francescobrivio committed:
changing edm::LogWarning to edm::LogInfo

francescobrivio committed:
Revert "changing edm::LogWarning to edm::LogInfo"

This reverts commit 705630565ba43e9eebbbb8476c48f6bcc71ce8f8.

francescobrivio committed:
fix edm::LogWarnin to edm::LogInfo

francescobrivio committed:
Fix indentation

cmsbuild commented:
A new Pull Request was created by  @francescobrivio  for master.

It involves the following packages:

Calibration/TkAlCaRecoProducers
DQM/BeamMonitor
RecoVertex/BeamSpotProducer


@perrotta, @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @slava77, @mmusich, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @makortel, @felicepantaleo, @GiacomoSguazzoni, @tocheng, @VinInn, @rovere, @ebrondol, @threus, @dgulhan this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

arunhep commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ae38b02bb1e784f8f9aa9635c2f32cf95d6d5daf
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18755/19858/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18755/19858/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 1659 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1829428
 - DQMHistoTests: Total failures: 40314
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1788934
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

rmanzoni commented:
hi,

this PR has 91X in its title and the branch to merge has 91X in its name as well, however it is definitely meant to be in place for the data taking, that is it should enter 92X.

@francescobrivio can you please change the title to 92X?

@franzoni @arunhep might be that this slipped under your radars because of the confusing name, but this is *the* PR for 2017 BS. I see Arun triggered the tests, they went well, please have a look and give +1 (or -1 of course, if that's the case) to move forwards

francescobrivio commented:
@rmanzoni Title changed to 92X

ghellwig commented on RecoVertex/BeamSpotProducer/src/PVFitter.cc:
```diff
@@ -103,51 +93,44 @@ PVFitter::~PVFitter() {
+              }
+              if (sumPt < minSumPt_) continue;
+          }
+          catch (...)
```
@francescobrivio this is forbidden by the CMS code rules
Please catch only the exception you want to handle here.

ghellwig commented on RecoVertex/BeamSpotProducer/src/PVFitter.cc:
```diff
@@ -401,12 +373,12 @@ bool PVFitter::runFitter() {
-      upar.Add("ez"    , 1.   	    , 0.1   	     , 0.   	    , 30. 	    ); // 8
+      upar.Add("ex"    , 0.010	    , 0.0001  	 , 0.   	, 10. 	    ); // 3
+      upar.Add("corrxy", 0.   	    , 0.02  	 , -1.  	, 1.  	    ); // 4
+      upar.Add("ey"    , 0.010	    , 0.0001  	 , 0.   	, 10. 	    ); // 5
```
The PR description says that nothing changes wrt. the previous situation, but here you are changing hard-coded parameter values.
Please clarify the impact of those.

ghellwig reviewed (changes requested):
@francescobrivio can you point me to the lines of code or configuration that fix the issue mentioned in https://github.com/cms-sw/cmssw/pull/18201#issuecomment-295599218.
Unfortunately the link from David is now dead as the logs are cleaned up in the meantime.

perrotta commented:
@francescobrivio 
- Please remove commented out lines from PVFitter.h, as you already did in previous #18201
- Please check otherwise if the track collection exists, as in the first comment of https://github.com/cms-sw/cmssw/pull/18755#pullrequestreview-39418364
- Please either explain or revert the changes in the hard coded parameters, as in the second comment of https://github.com/cms-sw/cmssw/pull/18755#pullrequestreview-39418364

franzoni commented:
hello @francescobrivio 
perhaps you may answer @ghellwig 's queries ?
We're preparing the PR's for our weekly ORP report and it would be good to proceed.

francescobrivio commented:
hi @franzoni sorry for the delay. I'm working on it right now.

francescobrivio committed:
Remove commented lines in PVFitter.h; revert hard coded parameters in PVFitter.cc

francescobrivio commented:
hi @perrotta and @ghellwig 
I removed the commented lines in PVFitter.h and reverted the hard coded parameters in PVFitter.cc,
I just have to fix the try/catch part and then I will push the fixes.
Do you have any suggestion on how to catch a "ProductNotFound" exception in cmssw?

makortel commented:
@francescobrivio Check beforehand if `edm::Handle` is valid, or `Ref` is not null and is available. Here is an example how we nowadays check in the vertex validation code that the track collection referenced by the vertex collection is available in the `edm::Event` (and ignore the vertex collection if the track collection is missing)
https://github.com/cms-sw/cmssw/blob/master/Validation/RecoVertex/src/PrimaryVertexAnalyzer4PUSlimmed.cc#L1373-L1391

francescobrivio committed:
Replace try/catch with check on edm::handle

cmsbuild commented:
Pull request #18755 was updated. @perrotta, @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @slava77, @vanbesien, @davidlange6 can you please check and sign again.

francescobrivio commented:
Hi @makortel thanks a lot for the tip, it was very useful!

ghellwig commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

ghellwig commented:
@francescobrivio thanks for the fixes, but I believe you still owe me an answer to this:

> @francescobrivio can you point me to the lines of code or configuration that fix the issue mentioned in https://github.com/cms-sw/cmssw/pull/18201#issuecomment-295599218.
Unfortunately the link from David is now dead as the logs are cleaned up in the meantime.

francescobrivio commented:
sorry I forgot to answer you!
the fix is indeed the try/catch which is now a more correct check on the edm::handle.
I hope this answers your question, if not, please let me know!
________________________________________
Da: Gregor Mittag [notifications@github.com]
Inviato: martedì 23 maggio 2017 18.58
A: cms-sw/cmssw
Cc: Francesco Brivio; Mention
Oggetto: Re: [cms-sw/cmssw] CMSSW 92X PVselections (#18755)

@francescobrivio<https://github.com/francescobrivio> thanks for the fixes, but I believe you still owe me an answer to this:

@francescobrivio<https://github.com/francescobrivio> can you point me to the lines of code or configuration that fix the issue mentioned in #18201 (comment)<https://github.com/cms-sw/cmssw/pull/18201#issuecomment-295599218>.
Unfortunately the link from David is now dead as the logs are cleaned up in the meantime.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/18755#issuecomment-303465968>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AHddMaJURUL42nojo781d2-UKytIKo87ks5r8xBCgaJpZM4NbzTD>.

ghellwig commented:
+1

ghellwig commented:
thanks for the explanation

cmsbuild commented:
+1
Tested at: 7a559cea2f1b8e630aa420005b586e8ceb44aa3f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18755/20047/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18755/20047/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 1678 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1833092
 - DQMHistoTests: Total failures: 9371
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1823541
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

perrotta commented:
+1
As in #18201 (already merged and the reverted), this PR basically adds two extra parameters to be used for the fit of the BeamSpot: they can be used for tests, but they are kept by now at default values that fully reproduce the previous behavior of the algo.
With respect to the previous implementation, here a protection is added with a check on the presence of the track collection linked to the PV. Explicitely checked that the previous failing wf 1004 is now ending up correctly.

perrotta commented:
@francescobrivio : please change the title of this PR to something more descriptive. The previous "Adding the possibility to select the PVs before the BeamSpot fit" would be fine, for example.

francescobrivio commented:
@perrotta title changed

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar
